libs/vkd3d/resource.c

```diff
@@ -865,6 +865,20 @@ static bool vkd3d_is_linear_tiling_supported(const struct d3d12_device *device,
             && (image_info->samples & properties.sampleCounts);
 }
 
+static VkImageLayout vk_common_image_layout_from_d3d12_desc(const D3D12_RESOURCE_DESC *desc)
+{
+    if (desc->Flags & D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS)
+        return VK_IMAGE_LAYOUT_GENERAL;
+
+    if (desc->Flags & D3D12_RESOURCE_FLAG_DENY_SHADER_RESOURCE)
+        return VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
```

Comment:
Why depth stencil and not color?

Reply:
`D3D12_RESOURCE_FLAG_DENY_SHADER_RESOURCE` is only allowed in combination with `ALLOW_DEPTH_STENCIL` according to [MSDN](https://docs.microsoft.com/en-us/windows/win32/api/d3d12/ne-d3d12-d3d12_resource_flags).

Reply:
Curious, but ok.

Reply:
Done.